lib/experimental/editor-settings.php

```diff
@@ -28,6 +28,12 @@ function gutenberg_enable_experiments() {
 	if ( gutenberg_is_experiment_enabled( 'gutenberg-no-tinymce' ) ) {
 		wp_add_inline_script( 'wp-block-library', 'window.__experimentalDisableTinymce = true', 'before' );
 	}
+
+	// Support the previous location for the Site Icon settings. To be removed
+	// when the required WP core version for Gutenberg is >= 6.5.0.
+	if ( ! is_wp_version_compatible( '6.5' ) ) {
+		wp_add_inline_script( 'wp-block-editor', 'window.__experimentalUseCustomizerSiteLogoUrl = true', 'before' );
```

Comment:
Yeah that's a good point, stuff in `lib/experimental` doesn't get audited for removal when we update the lowest supported WP.

Reply:
> I'm wondering if this could have a better home in compat/6.5?

Good point. My main thinking in keeping it under `lib/experimental` was that `compat/6.5` typically houses things that are intended to be backported, and it's probably a good thing for all the `window.__experimental*` globals to be co-located.

That said, I'm happy to move it around. Did you have a particular file in mind? It looks like most of the current ones are there to match files in `wordpress-develop` 🤔

Reply:
Oh wait, we have a `compat/wordpress-6.5/compat.php` file — I reckon that might be a good home, I can move it there 👍

Reply:
I think ideally it would be a new `compat/wordpress-6.5/editor-settings.php` file, because none of the existing files are related in any way 😅 

I see compat folders as the place we put things we'll want to remove when we no longer support that version. It's not always stuff that's to be/has been backported, but anything that is version-specific. Really it's mostly for the sake of easier cleanup later!

Reply:
I think it's okay to add a new file, e.g., [lib/compat/wordpress-6.4/theme-previews.php](https://github.com/WordPress/gutenberg/blob/d1912448b9e02d54ccd2adc680557c13f4934697/lib/compat/wordpress-6.4/theme-previews.php#L14-L14)

Also maybe with a comment to say that it doesn't need backporting?

Reply:
`compat.php` already has this heading `* WordPress 6.5 compatibility functions.`, so I think that might be a good home for now.

Reply:
Thanks for the discussion, folks! I've moved the location over in https://github.com/WordPress/gutenberg/pull/59485/commits/b8d9ae2d1825815fcbe69fbebe1380bf035fc4ff

Let me know if you'd like any further changes! 🙂

Reply:
Retested. LGTM

Reply:
Looks good!